### PR TITLE
feat(DTFS2-7202): async createApp and typescript api tests (dtfs-central-api)

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Introduction :pencil2:
One of the known reasons for api test flakiness is that the SQL database connection can sometimes take longer to initialise than the test suite takes to run. This happens because `createApp.js` does not `await` the SQL database connection initialisation. This can be fixed by making the `createApp` function `async`.

Making the `createApp` function `async` breaks the api tests due to the way the api is created. This PR creates a `test-api.ts` file which exports an instance of a `TestApi` class which performs the same logic as the old `api-tests/api.js` file, but with support for the new `async` nature of `createApp`.

These changes should both make api tests easier to write and reduce flakiness for `dtfs-central-api`.

## Resolution :heavy_check_mark:
- Adds a new `test-api.ts` test helper
- Adds a new setup step to the api test jest config
- Re-enables and fixes the rate-limiting api test
- Moves the `utils` api tests to `src/utils` and converts them to unit tests
- Removes `api-test-users.js`

### CODE REVIEW

**Review by AlexBramhill — COMMENTED** [review]
lgtm -- i know youre still working on this

File: dtfs-central-api/api-tests/mocks/test-users/api-test-users.ts [hunk 1]
```diff
@@ -87,64 +90,64 @@ const testUsers = [
...
+      const response: PostUserResponse = await testApi.as().post(user).to('/v1/user');
+      return {
+        ...user,
+        _id: response.body._id,
+      };
+    }),
+  );
 
-  return finder;
+  return finder(loadedUsers);
```
[the remark this answers is not in the document]
Comment (JoshBinns2000) [reply dtfs-central-api/api-tests/mocks/test-users/api-test-users.ts]: Shame that it's gone now, but thank you :)

File: dtfs-central-api/api-tests/mocks/test-users/api-test-users.ts [hunk 2]
```diff
@@ -87,64 +90,64 @@ const testUsers = [
       useTFM: true,
     },
   },
-];
-
-let notYetInitialised = true;
-const loadedUsers = [];
+] as unknown as PortalUser[];
```
[the remark this answers is not in the document]
Comment (JoshBinns2000) [reply dtfs-central-api/api-tests/mocks/test-users/api-test-users.ts]: I have just introduced a new test helper called `aPortalUser` which could have been used here to make sure the type genuinely is `PortalUser`, but this file is gone now 🙃

[the remark this answers is not in the document]
Comment (JoshBinns2000) [reply dtfs-central-api/api-tests/mocks/test-users/api-test-users.ts]: The other point here is that I'm not really sure what type these users are meant to be. `PortalUser` is actually incorrect as they type is instead whatever the payload is for posting to the `/v1/user` endpoint, but that type isn't documented anywhere :/

File: trade-finance-manager-api/api-tests/v1/deals/deals-submit.api-test.js [hunk 3]
```diff
@@ -41,6 +41,15 @@ const getGefMandatoryCriteriaByVersion = jest.fn(() => Promise.resolve([]));
...
 
 describe('/v1/deals', () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date('2024-06-28'));
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
```
Comment (JoshBinns2000) [inline trade-finance-manager-api/api-tests/v1/deals/deals-submit.api-test.js:51]: This change should hopefully avoid the flakey result where the time-related field in the response can sometimes be off by around 1 ms

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**